corehq/apps/hqwebapp/widgets.py

```diff
@@ -158,3 +158,16 @@ def render(self, name, value, attrs=None, renderer=None):
                    id=html_attr(attrs.get('id', '')),
                    value=html_attr(self.params['value'] or '"{}"'.format(html_attr(value))),
                    options=html_attr(json.dumps([{'id': c[0], 'text': c[1]} for c in self.choices])))
+
+
+class GeoCoderInput(Input):
+
+    def __init__(self, attrs=None):
+        super(GeoCoderInput, self).__init__(attrs=attrs)
+
+    def render(self, name, value, attrs=None, renderer=None):
+        self.input_type = 'hidden'
+        output = super(GeoCoderInput, self).render(name, value, attrs, renderer)
+        return mark_safe("""
+            <div id="geocoder-proximity">{}</div>
```

Comment:
This is something of a nitpick, but it'd be better for this to be a class and for the js to loop over all elements with that class, rather than only allowing one geocoder on the page.

Reply:
exploring on this